main.tf

```diff
@@ -443,26 +436,48 @@ resource "null_resource" "confirm_network_healthy" {
   }
 }
 
+##############################################################################
+# Addons
+##############################################################################
+
 # Lookup the current default csi-driver version
 data "ibm_container_addons" "existing_addons" {
   cluster = local.cluster_id
 }
 
+locals {
+  # for each cluster, look for installed csi driver to get version. If array is empty (no csi driver) then null is returned
+  csi_driver_version = one([
+    for addon in data.ibm_container_addons.existing_addons.addons :
+    addon.version if addon.name == "vpc-block-csi-driver"
+  ])
+
+  # for each cluster in the clusters_map, get the addons and their versions and create an addons map including the corosponding csi_driver_version
+  cluster_addons = {
```

Comment:
this is a map that was only needed for SLZ because there are multiple clusters in the solution. You don't need this map here with id and resource_group.

Actually the SLZ code was borrowed from the existing code of this module, so you might not have had to change anything for the existing addon list. (I think the code lines 28-34 will work as-is)

Reply:
Revert few of the SLZ changes, but brought everything related to addons in a single place.